src/main/java/backend/resource/TurboLabel.java

```diff
@@ -4,153 +4,188 @@
     public static final String EXCLUSIVE_DELIMITER = ".";
     public static final String NONEXCLUSIVE_DELIMITER = "-";
 
-    private void ______SERIALIZED_FIELDS______() {
-    }
+    private final String fullName;
+    private final String shortName;
+    private final String groupName;
+    private final Grouping grouping;
 
-    private final String actualName;
     private final String colour;
-
-    private void ______TRANSIENT_FIELDS______() {
-    }
-
     private final String repoId;
 
-    private void ______CONSTRUCTORS______() {
-    }
-
-    public TurboLabel(String repoId, String name) {
-        this.actualName = name;
-        this.colour = "ffffff";
-        this.repoId = repoId;
-    }
 
     public TurboLabel(String repoId, String colour, String name) {
-        this.actualName = name;
+        this.fullName = name;
+
+        String[] splitted = splitKeyword(name);
+        this.groupName = splitted[0];
+        this.grouping = determineGrouping(splitted[1]);
+        this.shortName = splitted[2];
+
         this.colour = colour;
         this.repoId = repoId;
     }
 
-    public static TurboLabel nonexclusive(String repoId, String group, String name) {
-        return new TurboLabel(repoId, joinWith(group, name, false));
-    }
-
-    public static TurboLabel exclusive(String repoId, String group, String name) {
-        return new TurboLabel(repoId, joinWith(group, name, true));
+    public TurboLabel(String repoId, String name) {
+        this(repoId, DEFAULT_COLOR, name);
     }
 
     /**
      * Copy constructor
      */
     public TurboLabel(TurboLabel label) {
-        this.actualName = label.getActualName();
-        this.colour = label.getColour();
-        this.repoId = label.getRepoId();
+        this(label.getRepoId(), label.getColour(), label.getFullName());
     }
 
     public TurboLabel(String repoId, Label label) {
-        this.actualName = label.getName();
-        this.colour = label.getColor();
-        this.repoId = repoId;
+        this(repoId, label.getColor(), label.getName());
     }
 
     public TurboLabel(String repoId, SerializableLabel label) {
-        this.actualName = label.getActualName();
-        this.colour = label.getColour();
-        this.repoId = repoId;
+        this(repoId, label.getColour(), label.getFullName());
     }
 
-    private void ______METHODS______() {
+
+    /**
+     * Extracts group name, group delimiter and short name from a keyword
+     * group name is name of the group associated with a keyword
+     * short name is name of the keyword after omitting its group name and group delimiter
+     * Example: "priority.high" -> group name = "priority", group delimiter = ".", short name = "high"
+     * @param keyword
+     * @return String array of group name, group delimiter and short name
+     */
+    private static String[] splitKeyword(String keyword) {
+        Pattern p = Pattern.compile(String.format("^([^\\%s\\%s]+)(\\%s|\\%s)([^\\%s\\%s]*)",
```

Comment:
Is this regex supposed to match `keyword` with empty short name, such as `priority.`?

Btw, you can use
```
Pattern.compile(String.format("^([^\\%1$s\\%2$s]+)(\\%1$s|\\%2$s)([^\\%1$s\\%2$s]*)", EXCLUSIVE_DELIMITER, NONEXCLUSIVE_DELIMITER));
```
to avoid repeating the `EXCLUSIVE_DELIMITER` and `NONEXCLUSIVE_DELIMITER`.